### PR TITLE
Add better test coverage and mocks

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This PR further increases the test coverage.

### CODE REVIEW

File: app/backend/approaches/approach.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def serialize_for_results(self) -> dict[str, Any]:
         }
 
     @classmethod
-    def trim_embeddings(cls, embedding: Optional[List[float]]) -> Optional[str]:
+    def trim_embedding(cls, embedding: Optional[List[float]]) -> Optional[str]:
```
Comment (pamelafox) [inline app/backend/approaches/approach.py:56]: Renamed for clarity, I was getting confused by the plural in this function name

File: app/backend/approaches/approach.py [hunk 2]
```diff
@@ -190,8 +191,9 @@ async def compute_image_embedding(self, q: str, vision_endpoint: str, vision_key
         data = {"text": q}
 
         async with aiohttp.ClientSession() as session:
-            async with session.post(url=endpoint, params=params, headers=headers, json=data) as response:
-                response.raise_for_status()
+            async with session.post(
```
Comment (pamelafox) [inline app/backend/approaches/approach.py:194]: The raise_for_status option appears to achieve the same as calling raise_for_status, and it's easier on our mocks. :)

File: app/backend/approaches/chatreadretrievereadvision.py [hunk 3]
```diff
@@ -154,10 +154,9 @@ async def run_until_final_call(
             user_content.append({"text": "\n\nSources:\n" + content, "type": "text"})
         if include_gtpV_images:
             for result in results:
-                if result.sourcepage:
-                    url = await fetch_image(self.blob_container_client, result)
-                    if url:
-                        image_list.append({"image_url": url, "type": "image_url"})
+                url = await fetch_image(self.blob_container_client, result)
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrievereadvision.py:157]: The "if result.sourcepage" is repeated in fetch_image, so I removed the redundant check here.

File: app/backend/approaches/chatreadretrievereadvision.py [hunk 4]
```diff
@@ -183,7 +182,7 @@ async def run_until_final_call(
                 ThoughtStep(
                     "Generated search query",
                     query_text,
-                    {"semanticCaptions": use_semantic_captions, "vector_fields": vector_fields},
+                    {"use_semantic_captions": use_semantic_captions, "vector_fields": vector_fields},
```
Comment (pamelafox) [inline app/backend/approaches/chatreadretrievereadvision.py:185]: Cosmetic rename, to match the other key. That's what caused many of the snapshot diffs in this PR.

File: tests/conftest.py [hunk 5]
```diff
@@ -24,68 +23,37 @@
...
-        return self
-
-    async def __anext__(self):
-        if not self.data:
-            raise StopAsyncIteration
-        return AsyncPageIterator(self.data.pop(0))
-
-    def by_page(self):
-        return self
+from .mocks import (
```
Comment (pamelafox) [inline tests/conftest.py:26]: I moved most mocks into mocks.py because conftest.py was getting hard to wade through.

File: tests/conftest.py [hunk 6]
```diff
@@ -24,68 +23,37 @@
...
 
+@pytest.fixture
+def mock_compute_embeddings_call(monkeypatch):
+    def mock_post(*args, **kwargs):
+        if kwargs.get("url").endswith("computervision/retrieval:vectorizeText"):
+            return mock_computervision_response()
+        else:
+            raise Exception("Unexpected URL for mock call to ClientSession.post()")
+
+    monkeypatch.setattr(aiohttp.ClientSession, "post", mock_post)
```
Comment (pamelafox) [inline tests/conftest.py:54]: We need to be careful of mocking ClientSession.post() as we use it in two places, for computervision and for ADLS. Fortunately their tests don't use same mocks, but could bite us in future.

**Review by srbalakr — APPROVED** [review]

**Comment by srbalakr** [comment]
Hi @pamelafox I am planning to merge this repo to main repo tomorrow morning. Let me know if any planned item on this repo.

**Comment by pamelafox** [comment]
All done with my planned changes!

**Pull request merged**